### PR TITLE
Make it clearer that np.interp input must be monotonically increasing

### PR DESCRIPTION
**Pull request by dstansby**
xref https://github.com/numpy/numpy/issues/10448 - this is a subset of https://github.com/numpy/numpy/pull/17860 which only contains doc changes. Hopefully these make it clearer to users that they must supply monotonically increasing points to the function.

### CODE REVIEW

File: numpy/lib/function_base.py [hunk 1]
```diff
@@ -1337,8 +1337,8 @@ def interp(x, xp, fp, left=None, right=None, period=None):
     --------
     scipy.interpolate
 
-    Notes
-    -----
+    Warnings
```
Comment (rgommers) [inline numpy/lib/function_base.py:1340]: We don't use it a lot, but this is a valid section header in numpydoc, and in this case I agree it's warranted.

**Review by rgommers — APPROVED** [review]
LGTM, thanks @dstansby

File: numpy/lib/function_base.py (hunk 1 above)
Comment (eric-wieser) [reply numpy/lib/function_base.py:1340]: Isn't this section header supposed to be about `Warning` objects emitted via the `warnings` module?

Comment (rgommers) [reply numpy/lib/function_base.py:1340]: The numpydoc description of this section is _"An optional section with cautions to the user in free text/reST."_.

**Pull request merged**